### DISCUSSION TITLE
Webpack Errors: invalid string length

### DISCUSSION THREAD

**Discussion by cmeffley** [nss-evening-cohort-14/evening-client-side#99] [Help Me!] [answered]
Chris Meffley

Other tickets? None

I have fixed all the errors that were showing in my terminal, but when I try the entering a name in the sorting hat I keep getting the error code:
Uncaught RangeError: invalid string length.
The error is on the empty card string in the cardCreator function

https://stackoverflow.com/questions/28922057/uncaught-rangeerror-invalid-string-length-when-appending-to-a-string-in-javascr
https://teamtreehouse.com/community/getting-uncaught-rangeerror-invalid-string-length-at-this-part-statushtml-li-classout
https://stackoverflow.com/questions/46615619/uncaught-rangeerror-invalid-string-length

I've tried changing the i++ and putting spaces in the empty string, but I'm at a loss at this point. Also, my styling is off too, so hopefully, once this code resolve that does too.

**Comment by drteresavasquez** [comment]
@cmeffley jump in Hackery 2 with me so we can check it out

**Comment by cmeffley** [comment, marked as answer]
In the for loop under let card = ' ''; I changed it to:
for (let i = 0;  i < array.length; i) because of an earlier error message, so the loop was no incrementing. I put the i++ back, but the error returned. If I hovered my mouse over the red squiggly line, there was a link to an ESLint site (https://eslint.org/docs/rules/no-plusplus) that helped with the answer. I needed to put i += 1. There must be spaces between everything too!

**Reply by drteresavasquez** [reply]
@cmeffley was there perhaps something else you could have done to fix this? Maybe use an array method to loop instead of a for loop?

**Reply by cmeffley** [reply]
I could have changed to a different loop like a forEach, and then I would not have needed the i ++ or i += 1  in this case. The forEach loop would have gone through each element of the array without needing to write the i += 1.
